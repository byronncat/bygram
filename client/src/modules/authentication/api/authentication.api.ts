import axios, { AxiosResponse } from 'axios';
import { uri, API } from '@global';
import { AuthenticationInformation } from '../types';

export async function loginAPI(data: AuthenticationInformation): Promise<API> {
  return await axios
    .post(uri.getHostingServer('login'), data)
    .then((res: AxiosResponse) => res.data)
    .catch((err: any) => err.response.data);
}

export async function registerAPI(
  data: AuthenticationInformation,
): Promise<API> {
  return await axios
    .post(uri.getHostingServer('register'), data)
    .then((res: AxiosResponse) => res.data)
    .catch((err: any) => err.response.data);
}

export async function authenticateAPI(): Promise<API> {
  return await axios
    .get(uri.getHostingServer('authenticate'))
    .then((res: AxiosResponse) => res.data)
    .catch((err: any) => err.response.data);
}

export async function logoutAPI(): Promise<API> {
  return await axios
    .delete(uri.getHostingServer('logout'))
    .then((res: AxiosResponse) => res.data)
    .catch((err: any) => err.response.data);
}
