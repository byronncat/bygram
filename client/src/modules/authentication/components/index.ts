export { default as Brand } from './brand.component';
export { default as Divider } from './divider.component';
export { default as Form } from './form.component';
export { default as NavigationText } from './navigation-text.component';
