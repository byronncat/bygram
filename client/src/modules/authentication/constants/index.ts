import { FormFieldProps } from '../types'

export * as DEFAULT_VALUES from './default.constant'
export * as FIELD from './field.constant'

export const REGISTER_FIELD = [
  {
    name: 'email',
    type: 'email',
    placeholder: 'Email',
    validation: {
      required: 'Email is required',
      pattern: {
        value: /\S+@\S+\.\S+/,
        message: 'Entered value does not match email format',
      },
    },
  },
  {
    name: 'username',
    type: 'text',
    placeholder: 'Username',
    validation: {
      required: 'Username is required',
      minLength: {
        value: 3,
        message: 'Username must be at least 3 characters',
      },
    },
  },
  {
    name: 'password',
    type: 'password',
    placeholder: 'Password',
    validation: {
      required: 'Password is required',
      minLength: {
        value: 6,
        message: 'Password must be at least 6 characters',
      },
    },
  },
] as FormFieldProps[]

export const SEND_EMAIL_FIELD = [
  {
    name: 'email',
    type: 'email',
    placeholder: 'Email',
    validation: {
      required: 'Email is required',
      pattern: {
        value: /\S+@\S+\.\S+/,
        message: 'Entered value does not match email format',
      },
    },
  },
] as FormFieldProps[]
