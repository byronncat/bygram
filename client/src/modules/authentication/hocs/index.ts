export { default as AuthenticationHoc } from './authentication.hoc';
export { default as LandingHoc } from './landing.hoc';
export * from './landing.hoc';
