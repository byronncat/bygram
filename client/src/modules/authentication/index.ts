export * from './components'
export * from './constants'
export * from './hocs'
export * from './layouts'
export * from './providers'

export { default as LoginPage } from './pages/sign-in.page'
export { default as RegisterPage } from './pages/sign-up.page'
export { default as ForgotPasswordPage } from './pages/forgot-password.page'
export { default as ResetPasswordPage } from './pages/reset-password.page'
export * from './types/index.d'
