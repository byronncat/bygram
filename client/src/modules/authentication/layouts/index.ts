export { default as LandingPageLayout } from './landing-page.layout';
