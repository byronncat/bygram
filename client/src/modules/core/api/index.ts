export * from './post.api';
export * from './profile.api';
