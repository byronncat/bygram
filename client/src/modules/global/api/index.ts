export function getURLServer(path: string): string {
  return `${process.env.REACT_APP_API_URL || 'http://192.168.1.4:5000'}${path}`;
}

export function formatImageCDN(path: string, format: string): string {
  const urlParts = path.split('/');

  for (let i = 0; i < urlParts.length; i++) {
    if (urlParts[i].includes('upload')) {
      urlParts[i + 1] = format;
      path = urlParts.join('/');
      break;
    }
  }
  console.log(path);
  return path;
}
