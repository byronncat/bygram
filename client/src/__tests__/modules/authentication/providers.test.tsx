import {
  useClassNameContext,
  AuthenticationProvider,
  useAuthenticationContext,
  ClassNameProvider,
} from '@authentication';
import { render, renderHook, waitFor } from '@testing-library/react';
import { ReactProps } from '@global';
import { authenticateAPI } from '@/modules/authentication/api';
import { useEffect } from 'react';

jest.mock('@/modules/authentication/api');
describe('provider', () => {
  describe('className', () => {
    it('should return className object', async () => {
      const wrapper = ({ children }: ReactProps) => (
        <ClassNameProvider>{children}</ClassNameProvider>
      );
      const { result } = await renderHook(() => useClassNameContext(), {
        wrapper,
      });

      expect(result.current.className).toMatchSnapshot();
    });
  });

  // The issue might be related to the scope of the mock.
  describe('authentication', () => {
    beforeEach(() => {
      (authenticateAPI as jest.Mock).mockResolvedValue({
        success: true,
        message: '',
      });
    });

    it('should return isAuthenticated object', async () => {
      let result: boolean = false;
      const Test = () => {
        const { isAuthenticated } = useAuthenticationContext();

        useEffect(() => {
          result = isAuthenticated;
        }, [isAuthenticated]);
        return (
          <div data-testid="authentication">{isAuthenticated.toString()}</div>
        );
      };
      render(
        <AuthenticationProvider>
          <Test />
        </AuthenticationProvider>,
      );

      await waitFor(() => {
        expect(result).toBe(true);
      });
    });
  });
});
