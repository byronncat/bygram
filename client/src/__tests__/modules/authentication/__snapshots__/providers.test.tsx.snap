// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Providers ClassName should return className object 1`] = `
Object {
  "form": "mw-100 rounded",
  "formErrorMessage": "error-message mt-1 mb-2 fw-bolder",
  "formErrorMessageAnimation": "flicker-one",
  "formField": "form-field form-floating",
  "formInput": "form-input form-control",
  "formLabel": "form-label",
}
`;
